### PR TITLE
Remove/nuget.exe

### LINKED ISSUE #1: Move to Nuget.Client
We currently wrap nuget.exe and require `NUGET` environment variable to be set.

### LINKED ISSUE #2: X-Platform support
The current code relies heavily on `PATH` and `.exe` 

This needs some updating so that we can the diff command on linux/osx as well

### PR DESCRIPTION
**Pull request by Mpdreamz**
* updates to the latest justassembly/justdecompile
* Now installs nuget packages programatically (no reference to `nuget.exe` needed).
* Adds a `previous-nuget` source now that resolves the previous released nuget package to the version specified (whole fully inspired by https://github.com/elastic/elasticsearch-net/blob/master/build/scripts/Differ.fs#L22)
* On non windows systems assumes `git` is available rather than looking for `git.exe`

Closes #1 
Closes #2

### CODE REVIEW

File: AssemblyDiffer/Program.cs [hunk 1]
```diff
@@ -82,8 +81,9 @@ private static void Main(string[] args)
 					throw new Exception($"No exporter for format '{_format}'");
 
 				var exporter = exporters[_format];
+				var pairs = CreateAssemblyPairs(firstProvider, secondProvider).ToList();
```
Comment (russcam) [inline AssemblyDiffer/Program.cs]: Does it need `ToList()`?

File: AssemblyDiffer/Program.cs [hunk 2]
```diff
@@ -104,12 +104,17 @@ private static void Main(string[] args)
 			}
 		}
 
-		private static IEnumerable<AssemblyDiffPair> CreateAssemblyPairs(IAssemblyProvider firstProvider, IAssemblyProvider secondProvider) =>
-			firstProvider.GetAssemblies(_targets).Join(secondProvider.GetAssemblies(_targets),
+		private static IEnumerable<AssemblyDiffPair> CreateAssemblyPairs(IAssemblyProvider firstProvider, IAssemblyProvider secondProvider)
+		{
+			var first = firstProvider.GetAssemblies(Targets).ToList();
```
Comment (russcam) [inline AssemblyDiffer/Program.cs]: does it need `.ToList()`?

File: AssemblyDiffer/Program.cs [hunk 3]
```diff
@@ -104,12 +104,17 @@ private static void Main(string[] args)
...
+			var second = secondProvider.GetAssemblies(Targets).ToList();
```
Comment (russcam) [inline AssemblyDiffer/Program.cs]: does it need `.ToList()`?

File: AssemblyDiffer/Providers/NuGet/NuGetPackage.cs [hunk 4]
```diff
@@ -10,6 +10,10 @@ namespace Differ.Providers.NuGet
 {
 	public class NuGetPackage : NuGetPackageBase
 	{
+		public static NuGetPackage Skip { get; } = new NuGetPackage();
```
Comment (russcam) [inline AssemblyDiffer/Providers/NuGet/NuGetPackage.cs]: Would `None` or `Null` be more descriptive?

File: AssemblyDiffer/Providers/PreviousNuget/NuGetAssemblyProvider.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
+﻿using Differ.Providers.NuGet;
+
+namespace Differ.Providers.PreviousNuGet
+{
+	public class PreviousNuGetAssemblyProviderFactory : NuGetAssemblyProviderFactory
```
Comment (russcam) [inline AssemblyDiffer/Providers/PreviousNuget/NuGetAssemblyProvider.cs:5]: ++ to including in the differ

**Review by russcam — APPROVED** [review]
LGTM. Just left some small comments

**Comment by russcam** [comment]
The README should also be updated to include `previous-nuget` as an assembly provider.

Do you think each assembly provider should also have a `--help` ? Might be easier to do this with `CommandLine`

File: AssemblyDiffer/Providers/NuGet/NuGetPackage.cs (hunk 4 above)
Comment (Mpdreamz) [reply AssemblyDiffer/Providers/NuGet/NuGetPackage.cs]: Went with `NotFound`, naming things.. :)

**Comment by Mpdreamz** [comment]
> Do you think each assembly provider should also have a --help ? Might be easier to do this with CommandLine

That would be good, going to create an issue for it to follow up on.

**Pull request merged**